src/controllers/validation/locations.js

```diff
@@ -33,8 +45,9 @@ export default {
       pageNumber: Joi.number(),
       pageSize: Joi.number(),
       age: Joi.number(),
+      sortBy: Joi.string(), // TODO: specify three possible values
     })
-      .and('radius', 'latitude', 'longitude')
+      .and('latitude', 'longitude')
```

Comment:
If I'm not mistaken, you could still express the now-asymmetrical condition with Joi too, and avoid at least 1 of the custom checks:
```suggestion
      .with('radius', ['latitude', 'longitude'])
```


And you could probably also represent the `sortBy === SORT_BY_PROXIMITY_OPTION` condition and avoid that check too, e.g. something like:
```
  latitude: Joi.number().when('sortBy', { is: SORT_BY_PROXIMITY_OPTION, then: Joi.required() }),
  longitude: Joi.number().when('sortBy', { is: SORT_BY_PROXIMITY_OPTION, then: Joi.required() }),
```
or possibly at the end of the object:
```
      .when("sortBy", {
        is: SORT_BY_PROXIMITY_OPTION,
        then: Joi.object({
          latitude: Joi.number().required(),
          longitude: Joi.number().required(),
        }),
      })
```
But up to you whether you think this one's still clearer than the custom check (I do like the idea of keeping as much validation as possible in the validation files in general, but only assuming you find it readable).

Reply:
Fixed in 18660fae8dca4883ba368d0f46db3413b3c482cb

Validated:

![Screenshot from 2024-09-16 16-28-20](https://github.com/user-attachments/assets/2234b946-16ce-4535-8012-0338a6760f3e)
![Screenshot from 2024-09-16 16-28-10](https://github.com/user-attachments/assets/ac3dbbaf-62b5-4e76-b1f6-9d0cf3130572)
![Screenshot from 2024-09-16 16-20-46](https://github.com/user-attachments/assets/02215e6d-9f4a-451e-9cab-995072fbb813)
![Screenshot from 2024-09-16 16-20-34](https://github.com/user-attachments/assets/a4530329-0a4a-4539-8eae-be88176ffcac)